schim_client.bal

```diff
@@ -75,29 +75,41 @@ final http:Client schimClientEp = check new(schimEndpoint,
 );
 
 isolated function getDonors(int pageNumber, int pageCount) returns Donor[]|error {
-    int startIdx = pageNumber == 0 ? 1: pageCount + (pageNumber - 1 * pageCount);
-    SchimUserSearchRequest requestPayload = {
-        filter: "groups eq donor",
-        domain: userDomain,
-        startIndex: startIdx,
-        count: pageCount
-    };
-    SchimUserResponse schimResponse = check schimClientEp->post("/Users/.search", requestPayload);
-    User[]? availableUsers = schimResponse?.Resources;
-    if availableUsers is User[] {
-        return
-        from var {id, userName, name, emails} in availableUsers
-        select {
-            id: id, 
-            userName: userName, 
-            firstName: name.givenName, 
-            lastName: name.familyName, 
-            email: check getPrimaryEmail(emails)
-        };
-    }
+    // int startIdx = pageNumber == 0 ? 1: pageCount + (pageNumber - 1 * pageCount);
+    // SchimUserSearchRequest requestPayload = {
+    //     filter: "groups eq donor",
+    //     domain: userDomain,
+    //     startIndex: startIdx,
+    //     count: pageCount
+    // };
+    // SchimUserResponse schimResponse = check schimClientEp->post("/Users/.search", requestPayload);
+    // User[]? availableUsers = schimResponse?.Resources;
+    // if availableUsers is User[] {
+    //     return
+    //     from var {id, userName, name, emails} in availableUsers
+    //     select {
+    //         id: id, 
+    //         userName: userName, 
+    //         firstName: name.givenName, 
+    //         lastName: name.familyName, 
+    //         email: check getPrimaryEmail(emails)
+    //     };
+    // }
     return [];
 }
 
+type DonorScimResponse record {
+    Donor[] Resources;
+};
+
+isolated function getDonor(string donorId) returns Donor?|error {
+    DonorScimResponse|error scimResult = check schimClientEp->get("/Users?filter=id eq " + donorId + "&attributes=displayName");
```

Comment:
Is `displayName` available for all the users ? AFAIK this is again an optional field for SCHIM user object.

Reply:
We only name a `orgName` form the frontend. This is the closest we have. Other option is to add a `orgName` to all users. I when with this since it's already there. Anyway we'll have to manually add name to each users (I think, need to clarify with sherazad)